contracts/CreditVoucher.sol

```diff
@@ -154,13 +158,23 @@ contract CreditVoucher is ICreditVoucher, ERC721Enumerable, Pausable, EIP712, Ow
 
         voucher.redeemedAt = current;
         voucher.redeemed = true;
+        totalCredits -= voucher.creditValue;
         _burn(_tokenId);
 
         IERC20(USDC).transfer(msgSender, voucher.creditValue);
 
         emit Redeemed(msgSender, _tokenId);
     }
 
+    /**
+     * @notice Burn a voucher
+     * @dev    Caller can be ANYONE
+     * @param _tokenId The tokenId to be redeemed
+     */
+    function burn(uint256 _tokenId) public {
```

Comment:
This operation should only succeed if it is called by the owner of the token or if the token has expired.